packages/block-editor/src/components/use-block-sync/index.js

```diff
@@ -199,6 +246,7 @@ export default function useBlockSync( {
 			) {
 				pendingChanges.current.outgoing = [];
 			}
+			// TODO: problematic if getBlocks has alternate client IDs
 		} else if ( getBlocks( clientId ) !== controlledBlocks ) {
```

Comment:
Need to fix this check. Theoretically, if blocks are being mapped to local IDs, then `getBlocks` will return one array with local IDs, but the controlledBlocks will use the IDs which come from the entity. As a result, the references will always be different.

Reply:
I don't think this is causing any bugs, though. It's probably just here for performance